### PR TITLE
Update training data sales sample and outlier flags

### PR DESCRIPTION
**Pull request by dfsnow**
This PR makes three major changes to the sales sample used to train the condominium model:

1. It excludes all sales of "nonlivable" spaces from the training data. Nonlivable spaces include parking, storage areas, common areas, etc. These typically have much lower values than actual condominium units. They are also valued using a separate methodology (see #25). As such, they should not be included in the training data.
2. It _includes_ select "multi-sales". These are sales that involve more than one PIN. In the case of condos, a multi-sale typically involves a unit and a deeded parking space. These sales are important to include because they make up a significant minority of sales (~ 25%). Some buildings _only_ have multi-sales.
3. It adjusts the multi-sale sale prices to capture _only the value of the livable unit_. It does this by taking the proportion of % of ownership between the unit and parking PINs, then applying it to the total sale price. This is necessary because the condo model only values livable PINs, so including the full sale price (unit + parking) will bias the model upward.

### CODE REVIEW

File: params.yaml [hunk 1]
```diff
@@ -146,6 +146,7 @@ model:
     all:
       - "meta_township_code"
       - "meta_nbhd_code"
+      - "meta_tieback_proration_rate"
```
Comment (dfsnow) [inline params.yaml:149]: We accidentally dropped this on one of the earlier PRs, but it needs to be included. It's the most important single predictor in the model.

File: params.yaml [hunk 2]
```diff
@@ -159,10 +160,7 @@ model:
       - "loc_longitude"
       - "loc_latitude"
       - "loc_census_tract_geoid"
-      - "loc_env_flood_fema_sfha"
       - "loc_env_flood_fs_factor"
-      - "loc_env_flood_fs_risk_direction"
-      - "loc_env_airport_noise_dnl"
```
Comment (dfsnow) [inline params.yaml:165]: Following the lead of the res model, I'm kicking out underperforming variables that are already largely covered by other vars.

File: params.yaml [hunk 3]
```diff
@@ -294,27 +287,27 @@ model:
       # for setting this parameter is to discover a good fixed value using CV,
       # then manually set that value for non-CV runs (which don't use
       # early stopping by default)
-      num_iterations: 2500
-      learning_rate: 0.015
+      num_iterations: 2275
+      learning_rate: 0.011
```
Comment (dfsnow) [inline params.yaml:291]: These params are pulled from an earlier long CV run (`2024-02-11-gracious-manasi`) that yielded similar, but _slightly_ improved results.

File: params.yaml [hunk 4]
```diff
@@ -355,9 +348,9 @@ pv:
 
   # Rounding settings to apply to initial predictions. Rounding is done to
   # indicate to property owners that model values are estimates, not exact
-  round_break: [1000, 10000, 100000]
-  round_to_nearest: [1, 500, 5000, 10000]
-  round_type: "floor"
+  round_break: [1000, 10000]
+  round_to_nearest: [1, 10, 100]
+  round_type: "ceiling"
```
Comment (dfsnow) [inline params.yaml:353]: This is reverting an earlier mistake, the condo model should retain the same rounding from last year.

File: pipeline/00-ingest.R [hunk 5]
```diff
@@ -205,21 +206,57 @@ message("Adding time features and cleaning")
...
+    # If there are multiple PINs associated with a sale, take only the
+    # proportion of the sale value that is attributable to the main unit (based
+    # on percentage of ownership)
+    total_proration_rate = sum(meta_tieback_proration_rate, na.rm = TRUE),
+    meta_sale_price = as.numeric(meta_sale_price),
+    meta_sale_price = ifelse(
+      n() == 2 & keep_unit_sale,
+      meta_sale_price * (meta_tieback_proration_rate / total_proration_rate),
+      meta_sale_price
+    ),
```
Comment (dfsnow) [inline pipeline/00-ingest.R:227]: This is the most critical change in this PR, as its directly adjusting the outcome. Here we're trying to pull out the proportion of the total sale price attached to _only the unit_.

File: pipeline/00-ingest.R [hunk 6]
```diff
@@ -205,21 +206,57 @@ message("Adding time features and cleaning")
...
+        "Low price (multi)",
+      meta_sale_price > 1700000 & meta_sale_num_parcels == 2 ~
+        "High price (multi)",
+      TRUE ~ sv_outlier_type
+    ),
+    sv_is_outlier = ifelse(
+      meta_sale_price < 50000 & meta_sale_num_parcels == 2 |
+        meta_sale_price > 1700000 & meta_sale_num_parcels == 2,
+      TRUE,
+      sv_is_outlier
```
Comment (dfsnow) [inline pipeline/00-ingest.R:253]: I pulled these boundaries from looking at existing flags on the overall sales sample.

File: pipeline/00-ingest.R (hunk 5 above)
Comment (wrridgeway) [reply pipeline/00-ingest.R:227]: Could you just add a note that `sum(meta_cdu == "GR", na.rm = TRUE) == 1` is how we make sure 1 (and only 1) of the parcels being sold in a multisale is a parking space?

File: pipeline/00-ingest.R [hunk 7]
```diff
@@ -205,21 +206,57 @@ message("Adding time features and cleaning")
...
+  mutate(
+    sv_outlier_type = case_when(
+      meta_sale_price < 50000 & meta_sale_num_parcels == 2 ~
+        "Low price (multi)",
+      meta_sale_price > 1700000 & meta_sale_num_parcels == 2 ~
+        "High price (multi)",
+      TRUE ~ sv_outlier_type
+    ),
+    sv_is_outlier = ifelse(
+      meta_sale_price < 50000 & meta_sale_num_parcels == 2 |
```
Comment (wrridgeway) [inline pipeline/00-ingest.R:250]: I'm trusting we don't have to wrap the `&` statements in parentheses due to R's precedence of `&` before `|`

**Review by wrridgeway — APPROVED** [review]

File: pipeline/00-ingest.R (hunk 7 above)
Comment (dfsnow) [reply pipeline/00-ingest.R]: Clarified in 6d8055d

File: pipeline/00-ingest.R (hunk 5 above)
Comment (dfsnow) [reply pipeline/00-ingest.R:229]: Clarified in 6d8055d

**Pull request merged**